src/sortStudents.ts

```diff
@@ -1,16 +1,81 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: true;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(array: number[]): number {
+  const sum: number
+    = array.reduce((accum: number, current: number) => accum + current, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / array.length;
+}
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const copy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        return copy.sort(
+          (a: Student, b: Student) => a.name.localeCompare(b.name),
+        );
+      }
+      break;
+    case SortType.Surname:
+      if (order === 'asc') {
+        return copy.sort(
+          (a: Student, b: Student) => a.surname.localeCompare(b.surname),
+        );
+      }
+      break;
```

Comment:
you may join a few cases in this way:
```suggestion
    case SortType.Name:
    case SortType.Surname:
      if (order === 'asc') {
        return copy.sort(
          (a: Student, b: Student) => a[sortBy].localeCompare(b[sortBy]),
        );
      }
      break;
```
the same with Age and Married (use Married code, with transforming to `number`)
the only thing you have to do to make it possible is to add values in the enum which match with keys in `student` object, example:
```
enum Example {
  Name = 'name'
}
```